spinalcordtoolbox/reports/qc2.py

```diff
@@ -334,15 +337,47 @@ def sct_deepseg_axial(
     ) if fname_seg_lesion else None
     if fname_seg_lesion:
         img_seg_lesion.data = (img_seg_lesion.data > 0.5) * 1
+    img_qc_seg = resample_nib(
+        image=img_qc_seg,
+        image_dest=img_input,
+        interpolation='linear',
+    ) if fname_qc_seg else None
+
+    # Using -qc-seg mask if available, we remove slices which are empty (except for the first 3 and last 3 slices just around the segmented slices)
+    if fname_qc_seg:
+        # We dilate the mask to keep the first and last 3 slices
+        first_slice = min(np.where(img_qc_seg.data)[0]) - 3
+        last_slice = max(np.where(img_qc_seg.data)[0]) + 3
+        # Now we remove the slices from the input image, the spinal cord mask and the lesion mask
+        img_input.data = img_input.data[first_slice:last_slice]
+        img_seg_sc.data = img_seg_sc.data[first_slice:last_slice]
+        if img_seg_lesion:
+            img_seg_lesion.data = img_seg_lesion.data[first_slice:last_slice]
+        img_qc_seg.data = img_qc_seg.data[first_slice:last_slice]
```

Comment:
I notice that this section is duplicated between the two QC functions.

It might be a little simpler if we put the cropping code into a function, and then call it on the images:

```suggestion
    # Using -qc-seg mask if available, we remove slices which are empty
    for img_to_crop in [img_input, img_seg_sc, img_seg_lesion, img_qc_seg]:
        if fname_qc_seg and img_to_crop:
            img_to_crop.data = crop_with_mask(img_to_crop.data, img_qc_seg)
```

```python
def crop_with_mask(array, img_crop, pad=3)
    first_slice = min(np.where(img_crop.data)[0]) - pad
    last_slice = max(np.where(img_crop.data)[0]) + pad
    return array[first_slice:last_slice]
```

Reply:
Alternatively, would it be more straightforward to use `sct_crop_image` with the `-m` (mask) and `-dilate` options to achieve this same behavior?

Reply:
That makes sense!! We don't need a perfectly accurate cropped image since it's just for QC, so I agree with your logic. :)